cpp/src_prims/stats/histogram.h

```diff
@@ -236,6 +236,27 @@ void histogramImpl(HistType type, int* bins, IdxT nbins, const DataT* data,
   };
 }
 
+template <typename IdxT>
+HistType evaluateBestHistAlgo(IdxT nbins) {
+  size_t smem = maxSmem();
+  size_t requiredSize = nbins * sizeof(int);
+  if (requiredSize <= smem) {
+    return HistTypeSmem;
+  }
+  requiredSize = ceildiv<size_t>(nbins, 2) * sizeof(int);
+  if (requiredSize <= smem) {
+    return HistTypeSmemBits16;
+  }
+  requiredSize = ceildiv<size_t>(nbins, 4) * sizeof(int);
+  if (requiredSize <= smem) {
+    return HistTypeSmemBits8;
+  }
+  if (nbins % 4096 == 0) {
+    return HistTypeGmemSwizzle;
+  }
```

Comment:
You can use swizzle on any size, you just have to adjust the size to the next multiple of 4096.  Something like nbins = ((nbins + 4096)>>12) <<12;

Reply:
Hi Lars, since we also need to swizzle the histogram bins for this approach, I kept this algo for future optimization work. Will certainly make a note of this when we start working on this swizzle-based implementation. Thank you.